src/Storages/ColumnsDescription.cpp

```diff
@@ -869,6 +870,63 @@ bool ColumnsDescription::hasCompressionCodec(const String & column_name) const
     return it != columns.get<1>().end() && it->codec != nullptr;
 }
 
+void ColumnsDescription::setDefaultCompressionCodec(const String & codec_name)
```

Comment:
In general I am not sure if we need to rewrite the query. `CompressionReadBufferBase` [seems to read the compression codec from the compressed buffer](https://github.com/ClickHouse/ClickHouse/blob/9c10d4f402b4c83c250a5bb4367aaa24142a21e6/src/Compression/CompressedReadBufferBase.cpp#L120), thus I don't think rewriting the create query is necessary.

I think in this way it can also be ambigouos in some cases. I will explain this in the main comment of PR.